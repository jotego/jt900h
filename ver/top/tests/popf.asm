    ; PUSH F
    maxmode on
    relaxed on
    org 0
    ld a,0xbf    ; common header

    ld xsp, stack
    ld a,0xff
    push a
    pop f

    jp nc,bad_end
    jp nov,bad_end
    jp ne,bad_end

    ld a,0
    push a
    pop f

    jp c,bad_end
    jp ov,bad_end
    jp eq,bad_end

end_loop:
    ld hl,0xbabe
    ld (0xffff),0xff
    jp end_loop
bad_end:
    ld hl,0xdead
    ld (0xffff),0xff
    jp bad_end
data:
    dw 0xcafe,0xbeef,0xffff,0xeeee,0xcccc
    dw 0,0,0,0,0,0,0,0,0,0
stack:
    end