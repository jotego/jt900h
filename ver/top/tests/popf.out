00000000 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
-- index registers --
00000000 - 00000000 - 00000000 - 0000005e
SR = f800
